### PR TITLE
documentation: fix link to `installation-guide`, `README.agda`, `README.md`...

### PR DESCRIPTION
**Pull request by jamesmckinna**
Knock-on consequence(s) of #1743 / #2184

### CODE REVIEW

**Comment by gallais** [comment]
Grep tells me there are other occurences:

```shell
🐉:~/projects/agda-stdlib$ find . -type f | xargs grep "notes/"
./HACKING.md:[style-guide](https://github.com/agda/agda-stdlib/blob/master/notes/style-guide.md).
./HACKING.md:[style-guide](https://github.com/agda/agda-stdlib/blob/master/notes/style-guide.md).
./README.md:See the [installation instructions](https://github.com/agda/agda-stdlib/blob/master/notes/installation-guide.md) for the latest version of the standard library.
./doc/release-guide.txt:  - `notes/installation-guide.txt`
```

**Comment by jamesmckinna** [comment]
Thanks Guillaume! 
It would indeed seem that no good deed goes unpunished: my reward for suggesting we move things to `doc/` after your `README` PR... ;-)

File: doc/style-guide.md [hunk 1]
```diff
@@ -333,6 +333,16 @@ line of code, indented by two spaces.
   ...                  | false = filter p xs
   ```
 
+* Instance arguments, and their types, should use the vanilla ASCII/UTF-8 `{{_}}`
```
Comment (gallais) [inline doc/style-guide.md:336]: Are the unrelated changes meant to be included here?

Comment (jamesmckinna) [reply doc/style-guide.md:336]: Yes, at least AFAIAC, because it's miscellaneous documentation which has come up in the course of other PRs. Messy perhaps, and one reason I've been more considered (new label; new issue) about flagging such things now...

**Review by gallais — APPROVED** [review]
